oda_api/api.py

```diff
@@ -1339,3 +1340,48 @@ def from_response_json(cls, res_json, instrument, product):
                 p.meta_data = p.meta
 
         return d
+
+class ProgressReporter(object):
+    """
+    The class allows to report task progress to end user
+    """
+    def __init__(self):
+        self._callback = None
+        callback_file = ".oda_api_callback"  # perhaps it would be better to define this constant in a common lib
+        if not os.path.isfile(callback_file):
+            return
+        with open(callback_file, 'r') as file:
+            self._callback = file.read().strip()
+
+    @property
+    def enabled(self):
+        return self._callback is not None
+
+    def report_progress(self, stage: str=None, progress: int=50, substage: str=None, subprogress: int=None, message:str=None):
+        """
+        Report progress via callback URL
+        :param stage: current stage description string
+        :param progress: current stage progress in %
+        :param substage: current substage description string
+        :param subprogress: current substage progress in %
+        :param message: message to pass
+        """
+        callback_payload = {k: str(v) for k, v in locals().items() if v is not None and k != 'self'}
+
+        if not self.enabled:
+            logger.info('no callback registered, skipping')
+            return
+
+        logger.info('will perform callback: %s', self._callback)
+
+        if re.match('^file://', self._callback):
+            with open(self._callback.replace('file://', ''), "w") as f:
```

Comment:
Would be good to ensure that basedir is available and create the directory tree if not (even though "write callback to file" mode is only used in tests)

Reply:
Sorry, not sure about this at all. Not too good to create directories following a typo.

Reply:
Probably, you are right